src/visualization/types/Mosaic/options.tsx

```diff
@@ -62,13 +62,70 @@ const MosaicOptions: FC<Props> = ({properties, results, update}) => {
   }
 
   const xColumn = defaultXColumn(results.table, properties.xColumn)
-  const yColumn = mosaicYColumn(results.table, properties.ySeriesColumns?.[0])
+  const ySeriesColumns = defaultYSeriesColumns(
+    results.table,
+    properties.ySeriesColumns
+  )
+
+  const yLabelColumns = defaultYLabelColumns(
+    properties.yLabelColumns,
+    ySeriesColumns
+  )
 
   // TODO: make this normal DashboardColor[] and not string[]
   const colors = properties.colors.map(color => {
     return {hex: color} as Color
   })
 
+  const updateYLabelColumns = (
+    validYSeriesColumns: Array<string>,
+    currentYLabelColumns: Array<string>,
+    option: string
+  ): Array<string> => {
+    const columnExists = Array.isArray(currentYLabelColumns)
+      ? currentYLabelColumns.find(col => col === option)
+      : false
+    let updatedYLabelColumns = currentYLabelColumns || []
+
+    if (columnExists) {
+      updatedYLabelColumns = currentYLabelColumns.filter(fc => fc !== option)
```

Comment:
what is `fc`?

Reply:
i think originally copied/pasted from similar code for `fillColumns`.

Will change this to `currentYLabelColumn`